natlib/svcd.c

```diff
@@ -0,0 +1,187 @@
+        lua_newtable(L);
+        lua_settable(L, 3);
+        lua_pushstring(L, "handlers");
+        lua_newtable(L);
+        lua_settable(L, 3);
+        lua_pushstring(L, "ivkid");
+        lua_pushnumber(L, 0);
+        lua_settable(L, 3);
+        //Duplicate the TOS so the table is still there after
+        //setglobal
+        lua_pushvalue(L, -1);
+        lua_setglobal(L, "SVCD");
+#else
+    //Load the SVCD table that Lua created
+    //This will be index 3
+    lua_getglobal(L, "SVCD");
+    printf("Put table at %d\n", lua_gettop(L));
+#endif
+    //Now begins the part that corresponds with the lua init function
+
+    //SVCD.asock
+    lua_pushstring(L, "asock");
+    lua_pushlightfunction(L, libstorm_net_udpsocket);
+    lua_pushnumber(L, 2525);
+    lua_pushlightfunction(L, svcd_init_adv_received);
+    lua_call(L, 2, 1);
+    lua_settable(L, 3); //Store it in the table
+
+    //SVCD.ssock
+    lua_pushstring(L, "ssock");
+    lua_pushlightfunction(L, libstorm_net_udpsocket);
+    lua_pushnumber(L, 2526);
+    lua_pushstring(L, "wdispatch");
+    lua_gettable(L, 3);
+    lua_call(L, 2, 1);
+    lua_settable(L, 3); //Store
+
+    //SVCD.nsock
+    lua_pushstring(L, "nsock");
+    lua_pushlightfunction(L, libstorm_net_udpsocket);
+    lua_pushnumber(L, 2527);
+    lua_pushstring(L, "ndispatch");
+    lua_gettable(L, 3);
+    lua_call(L, 2, 1);
+    lua_settable(L, 3); //Store
+
+    //SVCD.wcsock
+    lua_pushstring(L, "wcsock");
+    lua_pushlightfunction(L, libstorm_net_udpsocket);
+    lua_pushnumber(L, 2528);
+    lua_pushstring(L, "wcdispatch");
+    lua_gettable(L, 3);
+    lua_call(L, 2, 1);
+    lua_settable(L, 3); //Store
+
+    //SVCD.ncsock
+    lua_pushstring(L, "ncsock");
+    lua_pushlightfunction(L, libstorm_net_udpsocket);
+    lua_pushnumber(L, 2529);
+    lua_pushstring(L, "ncdispatch");
+    lua_gettable(L, 3);
+    lua_call(L, 2, 1);
+    lua_settable(L, 3); //Store
+
+    //SVCD.subsock
+    lua_pushstring(L, "subsock");
+    lua_pushlightfunction(L, libstorm_net_udpsocket);
+    lua_pushnumber(L, 2530);
+    lua_pushstring(L, "subdispatch");
+    lua_gettable(L, 3);
+    lua_call(L, 2, 1);
+    lua_settable(L, 3); //Store
+
+    //manifest table
+    lua_pushstring(L, "manifest");
+    lua_newtable(L);
+    lua_pushstring(L, "id");
+    lua_pushvalue(L ,1);
+    lua_settable(L, -3);
+    lua_settable(L, 3);
+
+     //If id ~= nil
+    if (!lua_isnil(L, 1)) {
+        lua_pushlightfunction(L, libstorm_os_invoke_periodically);
+        lua_pushnumber(L, 3*SECOND_TICKS);
+        lua_pushlightfunction(L, libstorm_net_sendto);
+        lua_pushstring(L, "asock");
+        lua_gettable(L, 3);
+        //Pack SVCD.manifest
+        lua_pushlightfunction(L, libmsgpack_mp_pack);
+        lua_pushstring(L, "manifest");
+        lua_gettable(L, 3);
+        lua_call(L, 1, 1);
+        //Address
+        lua_pushstring(L, "ff02::1");
+        lua_pushnumber(L, 2525);
+        cord_dump_stack(L);
```

Comment:
Nevermind we found the implementation,
It is used for debugging (prints the entire stack to stdout)